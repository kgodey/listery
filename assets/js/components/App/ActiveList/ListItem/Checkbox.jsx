import PropTypes from 'prop-types'
import React from 'react'


const checkboxStyle = {
	display: 'inline-block',
	textAlign: 'center',
	width: '35px'
}


export const Checkbox = ({ checked, onClick }) => {
	return (
		<span style={checkboxStyle} className='pull-left'>
			<input type='checkbox' checked={checked} onClick={onClick} />
		</span>
	)
}

Checkbox.propTypes = {
	checked: PropTypes.bool.isRequired,
	onClick: PropTypes.func.isRequired
}
