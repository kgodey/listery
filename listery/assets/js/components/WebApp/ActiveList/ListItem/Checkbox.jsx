import PropTypes from 'prop-types'
import React from 'react'


const checkboxContainerStyle = {
	display: 'inline',
	width: '35px'
}

const checkboxStyle = {
	marginTop: '5px'
}

export const Checkbox = ({ checked, onClick }) => {
	return (
		<span style={checkboxContainerStyle} className='pull-left align-top'>
			<input type='checkbox' checked={checked} onChange={onClick} style={checkboxStyle} />
		</span>
	)
}

Checkbox.propTypes = {
	checked: PropTypes.bool.isRequired,
	onClick: PropTypes.func.isRequired
}
