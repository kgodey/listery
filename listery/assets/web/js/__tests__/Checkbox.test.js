import React from 'react'
import { Checkbox } from '../components/App/ActiveList/ListItem/Checkbox.jsx'
import renderer from 'react-test-renderer'


test('Checkbox renders correctly', () => {
	const handleCheckboxClick = jest.fn()
	const component = renderer.create(
		<Checkbox checked={true} onClick={handleCheckboxClick} />
	)

	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})
