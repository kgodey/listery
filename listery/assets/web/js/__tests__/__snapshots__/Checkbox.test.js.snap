// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<span
  className="pull-left align-top"
  style={
    Object {
      "display": "inline",
      "width": "35px",
    }
  }
>
  <input
    checked={true}
    onChange={[MockFunction]}
    style={
      Object {
        "marginTop": "5px",
      }
    }
    type="checkbox"
  />
</span>
`;
